src/mon/OSDMonitor.cc

```diff
@@ -5041,7 +5041,7 @@ epoch_t OSDMonitor::send_pg_creates(int osd, Connection *con, epoch_t next) cons
     return next;
   ceph_assert(!creating_pgs_by_epoch->second.empty());
 
-  MOSDPGCreate2 *m = nullptr;
+  MOSDPGCreate2 *m = new MOSDPGCreate2(creating_pgs_epoch);
```

Comment:
Can you consider make_message instead, which is (I think) the pref'ed interface today?